plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/expression/AggregateFunctionPatterns.java

```diff
@@ -85,4 +94,38 @@ private AggregateFunctionPatterns() {}
     {
         return Property.property("type", ConnectorExpression::getType);
     }
+
+    public static Pattern<List<Variable>> variables()
+    {
+        return new Pattern<>(Optional.empty())
+        {
+            @Override
+            public <C> Stream<Match> accept(Object object, Captures captures, C context)
+            {
+                if (!(object instanceof List)) {
+                    return Stream.of();
+                }
+                @SuppressWarnings("unchecked")
+                List<ConnectorExpression> arguments = (List<ConnectorExpression>) object;
+                if (!arguments.stream().allMatch(Variable.class::isInstance)) {
+                    return Stream.of();
+                }
+                return Stream.of(Match.of(captures));
+            }
+
+            @Override
+            public void accept(PatternVisitor patternVisitor)
+            {
+            }
+        };
+    }
+
+    public static Predicate<List<Variable>> variableTypes(Type... types)
+    {
+        requireNonNull(types, "types is null");
+        List<Type> variableTypes = ImmutableList.copyOf(types);
+        return variables -> variableTypes.equals(variables.stream()
+                .map(Variable::getType)
+                .collect(toImmutableList()));
```

Comment:
You can try to make this more generic:

```suggestion
    public static Predicate<List<? extends ConnectorExpression>> expressionTypes(Type... types)
    {
        List<Type> expectedTypes = ImmutableList.copyOf(requireNonNull(types, "types is null"));
        return expressions -> expressions.equals(variables.stream()
                .map(ConnectorExpression::getType)
                .collect(toImmutableList()));
```

but please check it will actually work with the upcoming code, ie that the more flexibility won't backfire somehow

Reply:
Thanks, it works for the purpose and I couldn't find any existing scenario where this flexibility will hurt.